### ISSUE TITLE
How do packets get out of the VPC?

### ISSUE THREAD

**Issue by kmkale** [QuinnyPig/tailscale-layer#1]
Hey Corey,
First of all thanks for publishing this. I could not find a explanation for 
`without having to use a NAT instance or Managed NAT Gateway to allow those functions to also speak to external resources.`
And could not find a comment link in your blog, so asking here. 
Without a NAT how do packets get out of the VPC for connections initiated from the Lambda?

**Comment by userhas404d** [comment]
The "magic" happens in the extension [here](https://github.com/QuinnyPig/tailscale-layer/blob/main/extension1.sh#L40) additional details on the options in that command [here](https://tailscale.com/kb/1112/userspace-networking/)

tl;dr: [SOCKS5](https://www.ipvanish.com/socks5-proxy/)

**Comment by quinnypig** [comment]
Apologies; I was unclear. Lambdas can either run in the "public" area, or (effectively) in a private subnet of yours. This extension lets you run the Lambdas in the usual public location, but traverse the Tailscale network to expose guarded resources in various locations, including public-but-locked-down subnets within AWS. 

tl;dr: If you have private subnets / resources in those private subnets that Lambda needs to talk to, you're still faced with a sad NAT choice.
